contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -413,6 +424,51 @@ get_logical_schema_name(const char *physical_schema_name, bool missingOk)
 	return logical_name;
 }
 
+int16
+get_dbid_from_physical_schema_name(const char *physical_schema_name, bool missingOk)
+{
+	Relation 	rel;
+	HeapTuple	tuple;
+	ScanKeyData	scanKey;
+	SysScanDesc	scan;
+	Datum		datum;
+	int16		dbid;
+	TupleDesc	dsc;
+	bool		isnull;
+
+	if (get_namespace_oid(physical_schema_name, false) == InvalidOid)
+		return NULL;
+
+	rel = table_open(namespace_ext_oid, AccessShareLock);
```

Comment:
why access share lock here? shouldn't we use row exclusive lock?

Reply:
No need for row exclusive lock as we use this function only for getting dbid using namespace_ext_oid relation. If you're considering DROP DB/SCHEMA case then we must be holding RowExclusiveLock while dropping.